Code fix

sobuj53 opened the pull request:
Code has been fixed by xerta555 Rom. Moved crond schedule to post boot entry.

sobuj53 committed:
Code fix

Code has been fixed by xerta555 Rom

sobuj53 committed:
move crond schedule statement

Crond schedule has been to line 26 of `service.sh` to wait for boot complete

sobuj53 committed:
making place for crond entry

Just some clean up make place for crond schedule entry.

sobuj53 committed:
move all variable

Moving all variable to compatibility.txt from service.sh

sobuj53 commented:
please don't merge yet. I think it has some issue.

xerta555 commented on system/bin/Detach:
```diff
@@ -221,7 +221,7 @@ rm -f "$MAGMOD/apps_lists.txt"
 # Adding app
 add() {
-SERVICESH=$MAGMOD/service.sh
+#SERVICESH=$MAGMOD/service.sh
```
The script can't run propeply as the var for the `services.sh` file is declarated for each functions to be sure that everyting will working in any situation.

xerta555 commented on system/bin/Detach:
```diff
@@ -435,7 +435,7 @@ esac
 # Removing app
 rem() {
-SERVICESH=$MAGMOD/service.sh
+#SERVICESH=$MAGMOD/service.sh
```
Same comment

xerta555 commented on compatibility.txt:
```diff
@@ -3,6 +3,9 @@
 ps_accounts=$("$MODDIR/sqlite" $PS_DATA_PATH "SELECT account FROM ownership" | sort -u | wc -l)
 ps_accounts_final=$((ps_accounts+1))
 
+MAGMOD=/data/adb/modules/Detach
```
I would prefere to have common vars at the beginning of the `compatibility.txt` file, to avoid any scripting issue(s)

sobuj53 committed:
Prevent recursive run of service.sh file

Because of this line `#sh "$MAGMOD/main.sh" > "$MAGMOD/first_detach_result.txt" 2>&1` in `compatibility.txt`, the `service.sh` file was running in infinite loop. So to mitigate the issue I created a separate `sh` file named `main.sh` which will be called by `service.sh` only once. I have tested this changes few times now and seems to work good. But need additional testing.

sobuj53 committed:
New Detach for testing

instead of writing to `service.sh` it will now write to `main.sh`

sobuj53 committed:
Merge pull request #2 from sobuj53/sobuj53-test-1-1

New Detach for testing

sobuj53 committed:
version bump

sobuj53 commented:
Can you please take a look at [sobuj53-test-1](https://github.com/sobuj53/Detach/tree/sobuj53-test-1)? 
What is this test repo about?
- Well after you fixed the `compatibility.txt` file the `service.sh` was running in infinite loop because of `sh "$MAGMOD/service.sh" > "$MAGMOD/first_detach_result.txt" 2>&1` line in the service file. As it was misspelled like `services.sh` that's why we didn't observe this behavior earlier.
- I tried to fixed this behavior by creating a new `sh` file and calling it from `service.sh` only once.

sobuj53 committed:
Add files via upload

sobuj53 committed:
Add files via upload

sobuj53 committed:
Add files via upload

xerta555 commented:
@hoypolli I have check your main.sh file and I'm not necessarily against adding a new sh file in the module, but I've never had a negative feedback when the script doesn't run when the device starts 🤷🏼‍♂️

sobuj53 committed:
Add support for youtube music

This will enable built in support for yourtube music. To enable this support add `YouTube Music` line in `Detach.txt` or remove `#` sign in front of it.

sobuj53 committed:
Add built-in support for youtube music

This is to add built in support for youtube music. Remove `#` sign in front of `YouTube Music` and flash detach module to activate.

sobuj53 committed:
Youtube music support

This is to add built in support for youtube music. Remove `#` sign in front of `YouTube Music` and flash detach module to activate.

sobuj53 committed:
Detach txt update

Added built-in support for Youtube music

sobuj53 committed:
Update README.md

xerta555 commented on Detach.txt:
```diff
@@ -0,0 +1,45 @@
+#Phone 
+#Photos 
+YouTube
+#YouTube Music
```
It's useless to adding another app here if you don't edit the `customize.sh` file..

sobuj53 replied:
I did not updated here

sobuj53 replied:
in test repo. I'll push the change to main repo after completing the testing

xerta555 replied:
> in test repo. I'll push the change to main repo after completing the testing

Np. Take your time.

sobuj53 committed:
code fix

Fixed `su -c detach -a`, now user can add app on the fly via Terminal emulator. And various other code fixes.

sobuj53 committed:
placed all code in one place

loop ending code has been placed here  for easy maintenance rather than placing them in 3 other places

sobuj53 committed:
code improvement

sobuj53 committed:
offline list for existing detach app

The detach code will check this list for existing `Detach.txt` app rather getting names from online.

sobuj53 committed:
code improvement

Now detach will use fully dynamic `$MODDIR` variable, if magisk changes its mounting path, we'll only need to reflash detach module again

sobuj53 committed:
update

sobuj53 committed:
version bump

sobuj53 committed:
Update Detach.txt

sobuj53 committed:
Update Detach.txt

sobuj53 committed:
code fix

Fixed `su -c detach -a`, now user can add app on the fly via Terminal emulator. And various other code fixes.

sobuj53 committed:
typo fix

sleep 1

sobuj53 committed:
typo fix

sobuj53 committed:
typo fix

`sleep 1`

sobuj53 committed:
sed fix

code fix

sobuj53 committed:
code fix

sed command fix

sobuj53 committed:
Merge pull request #1 from sobuj53/sobuj53-test-1

various code fixes

sobuj53 committed:
remove unnecessary variable

sobuj53 committed:
typo

sobuj53 commented:
I think the pull request is ready to merge. Please test and merge if you think it's okay

sobuj53 committed:
variable fix

sobuj53 committed:
Existing app detection fixed

This fix will detect if an app is already listed in `main.sh`

sobuj53 committed:
typo fix

file name wasn't provided

sobuj53 committed:
android support update

sobuj53 committed:
Removed unnecessary instructions

Even if the App name is specified in `Detach.txt` the only way to detach an app without re-flashing module is to use `su -c detach -t` the previous instructions is invalid.

sobuj53 committed:
arm64-v8a

This is only suitable for `arm64-v8a` CPU. Also `sepolicy.rule` has been added to work fully under selinux enforcing system.

sobuj53 committed:
removed selinux permissive method

introduced sepolicy.rule to work under enforcing system.

sobuj53 committed:
database restoration removed

As it was broken anyway

sobuj53 committed:
Removed database restoration

Database restoration is broken so removed. PS will update its database anyway so...

sobuj53 committed:
extract sqlite3 based on cpu arc

This will extract sqlite3 based on CPU `$ARCH `

sobuj53 committed:
delete sqlite_arm64

will add all cpu arc sqlite3

sobuj53 committed:
sqlite3 has been updated

Update sqlite3 binary to it's current latest  `version 3.35.5`

sobuj53 commented:
@xerta555 can you please test [Detach-test](https://github.com/sobuj53/Detach/tree/sobuj53-test) repo? I've updated sqlite3 to its latest version 3.35.5. Also added magisk sepolicy rules to work under fully selinux enforcing system. We can merge those changes if you find them ok.

sobuj53 committed:
variable passing fixed

Restrict passing variable

sobuj53 committed:
typo fixed

sobuj53 committed:
get present directory fixed

sobuj53 committed:
Dynamic line detection for `#Other applications`

Removed static line (46) assignment for `# Other applications`. Now it can be detected anywhere in the `Detach.txt` file

sobuj53 committed:
Add files via upload

sobuj53 committed:
Merge pull request #3 from sobuj53/sobuj53-test

Sqlite3 binary update and code fix

sobuj53 committed:
version bump

version=v5 BETA 1

sobuj53 committed:
make sure detach list is at least 41 line long

sobuj53 committed:
comment in `Detach.txt` after removing app

This will put `#` sign after user remove an app using `su -c detach -r`

sobuj53 committed:
don't add #app from `Other applications`

This will prevent detach to add app name starting with `#` from `#Other applications`

sobuj53 committed:
some error fixes

sobuj53 committed:
Code fix

sobuj53 commented:
@xerta555 hope this will fix any surprise error. Hopefully those are my final changes

xerta555 commented:
> @xerta555 hope this will fix any surprise error. Hopefully those are my final changes

I also hope, i have in mind to test internally on Telegram the last BETA test version, then from 10 positive feedbacks, migrate all the changes of the module to make the brand new version official.

sobuj53 committed:
Copy Detach.txt to storage if not exists

This will copy `Detach.txt` to root of storage if not found during installation.

sobuj53 committed:
it was a mistake

Wrong changes to wrong file

sobuj53 committed:
Copy Detach.txt to storage if not exist

This will copy `Detach.txt` to root of storage if not found during installation.

sobuj53 committed:
comment and uncomment fix

This will fix app commenting (`#`) while detaching and un-detaching an app

sobuj53 committed:
code fix

sobuj53 committed:
abort installation if detach.txt is missing

This will copy `Detach.txt` to storage if not present and abort installation

sobuj53 committed:
ui print fix

sobuj53 committed:
variable fix

sobuj53 committed:
some typo fix

xerta555 commented on customize.sh:
```diff
@@ -107,7 +107,7 @@ if [ ! -e "$CONF" ]; then
 	echo -e "!!! Your Detach.txt file does not exist !!!"
 	echo -e "!!! Copying Detach file to your storage root!!!"
-	cp -af "$TMPDIR/Detach.txt" "$DetachFile"
+	cp -fr "$TMPDIR/Detach.txt" "$DetachFile"
```
Why using `-r` argument?

sobuj53 replied:
not really needed but it will make sure to overwrite the file if already there

xerta555 commented on customize.sh:
```diff
@@ -126,7 +126,7 @@ grep -q '\.' "$TMPDIR/SYN_CONF.txt"; if [ $? -eq 0 ]; then
 	echo -e "!- Your "$CONF" file contain error(s)"
 	ui_print "Now copying new Detach file, check and try again."
-	cp -afr "$TMPDIR/Detach.txt" "$DetachFile"
+	cp -fr "$TMPDIR/Detach.txt" "$DetachFile"
```
Same here

sobuj53 replied:
It's necessary in this case. As it's now necessary to overwrite already existing file.

sobuj53 committed:
Remove unnecessary space

sobuj53 committed:
check if user has new detach.txt file or not

sobuj53 committed:
Advice user on error

Added proper advice to user on error.

sobuj53 committed:
version bump
